modules/gapi/include/opencv2/gapi/util/util.hpp

```diff
@@ -97,6 +97,36 @@ namespace detail
     template <typename T1, typename... Ts>
     struct all_unique<T1, Ts...> : std::integral_constant<bool, !contains<T1, Ts...>::value &&
                                                                  all_unique<Ts...>::value> {};
+
+    template<typename>
+    struct tuple_wrap_helper;
+
+    template<typename T> struct tuple_wrap_helper
+    {
+        using type = std::tuple<T>;
+        static type get(T&& obj) { return std::make_tuple(std::move(obj)); }
+    };
+
+    template<typename... Objs>
+    struct tuple_wrap_helper<std::tuple<Objs...>>
+    {
+        using type = std::tuple<Objs...>;
+        static type get(std::tuple<Objs...>&& objs) { return std::forward<std::tuple<Objs...>>(objs); }
+    };
+
+    enum class PackageObjectTag {
+        KERNEL,
+        TRANSFORMATION
+    };
+
+    struct KernelTag
+    {
+    };
+
+    struct TransformTag
+    {
+    };
```

Comment:
These structures must be moved somewhere else

Reply:
Moved to gcommon.h